ome_zarr/format.py

```diff
@@ -12,6 +12,11 @@
 def format_from_version(version: str) -> "Format":
 
     for fmt in format_implementations():
+
+        # Support floating-point versions like `0.2`
+        if isinstance(version, float):
+            version = str(version)
```

Comment:
While this is primarily consumed at the reader level currently and hence quite downstream of the generation process, this raises the question of whether this library should also raise a `WARN` level logging statement indicating the mismatch.

Reply:
Possibly, though I'm already annoyed by the format mismatch warnings. Perhaps we could differentiate between a more purely user (`view()`) and developer (`validate()`) context.

Reply:
I am quite annoyed by the format mismatch warning as well - happy to create an issue or even look into it 😄 

In this scenario, the the warning is about a core issue with the data validation rather than exposing the internal guessing logic of the library so assuming I had to pick between the two, I would personally only keep the former. The idea of differentiating the usage contexts is interesting and possibly brings us back to the validation flag that @will-moore introduced while working on https://github.com/ome/ome-zarr-py/pull/189.

Reply:
I've found why it's duplicated and will open a PR. We'll still need to decide on the _first_ mismatch.